user_guide_src/source/installation/upgrade_models.rst

```diff
@@ -0,0 +1,71 @@
+Upgrade Models
+###############################################################################
+
+.. contents::
+    :local:
+    :depth: 1
+
+
+Documentations
+============================================================
+
+- `Model Documentation Codeigniter 3.X <http://codeigniter.com/userguide3/general/models.html>`_
+- `Model Documentation Codeigniter 4.X <http://codeigniter.com/user_guide/models/model.html>`_
+
+
+What has been changed
+============================================================
+- The CI4 model has much more functionality, including automatic database connection, basic CRUD, in-model validation, and automatic pagination
+- Since Namespaces has been added in CodeIgniter 4, the models must be changed to support namespaces.
+
+Upgrade Guide
+============================================================
+1. First, move all model files to the folder ``app/Models``
+2. Add this line just after the opening php tag: ``namespace App\Models;``
+3. Below the ``namespace App\Models;`` line add this line: ``use CodeIgniter\Model;;``
+4. Replace ``extends CI_Model`` with ``extends Model``
+5. Instead of CI3’s $this->load->model(x);, you would now use $this->x = new X();, following namespaced conventions for your component
```

Comment:
```suggestion
1. First, move all model files to the folder ``app/Models``.
2. Add this line just after the opening php tag: ``namespace App\Models;``.
3. Below the ``namespace App\Models;`` line add this line: ``use CodeIgniter\Model;``.
4. Replace ``extends CI_Model`` with ``extends Model``.
5. Instead of CI3’s ``$this->load->model(x);``, you would now use ``$this->x = new X();``, following namespaced conventions for your component. Alternatively, you can use the ``model`` function: ``$this->x = model('X');``.
```